agentendpoint/patch_windows.go

```diff
@@ -107,12 +108,16 @@ func (r *patchTask) wuaUpdates(ctx context.Context) error {
 		return err
 	}
 
-	// We keep searching for and installing updates until the count == 0 or there is an error.
-	retries := 20
-	for i := 0; i < retries; i++ {
+	// We keep searching for and installing updates until the count == 0,
+	// we get a stop signal, or retries exceed 10.
+	retries := 10
+	for i := 1; i <= retries; i++ {
+		if err := r.reportContinuingState(ctx, agentendpointpb.ApplyPatchesTaskProgress_APPLYING_PATCHES); err != nil {
+			return err
```

Comment:
Do we need to add/update tests foe this change?

Reply:
We currently can't unit test the system calls (WUA api), our e2e test handle the full flow but we don't exactly have a test for task cancelation right now. This change is strictly logging and moving the retry loop though.